### PR TITLE
Lets you choose the type of camo for PAS-11 and M-10 armor

### PR DESCRIPTION
**Pull request by CUST-0M**
## About The Pull Request

Allows you to choose the camo sprite for the PAS-11 armour vest and M-10 standard helmet. You do this by using it in hand, waiting 3 seconds and choosing whichever colour you want from the menu. 

## Why It's Good For The Game

Lets you be tacticool

### CODE REVIEW

File: code/modules/clothing/head/helmet.dm [hunk 1]
```diff
@@ -305,6 +305,28 @@
 /obj/item/clothing/head/helmet/marine/standard
 	flags_item_map_variant = (ITEM_JUNGLE_VARIANT|ITEM_ICE_VARIANT|ITEM_PRISON_VARIANT|ITEM_ICE_PROTECTION)
 
+/obj/item/clothing/head/helmet/marine/standard/attack_self(mob/user)
+
+	///List of colors for the M-10 and their respective icon states
+	var/static/list/helmet_colors = list(
```
Comment (BraveMole) [inline code/modules/clothing/head/helmet.dm:311]: what? why are you redeclaring the colors everytime.

Put that on the item, not in the proc


@@ -305,6 +305,28 @@
/obj/item/clothing/head/helmet/marine/standard
	flags_item_map_variant = (ITEM_JUNGLE_VARIANT|ITEM_ICE_VARIANT|ITEM_PRISON_VARIANT|ITEM_ICE_PROTECTION)
	var/static/list/helmet_colors = list( bla bla bla...

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by CUST-0M** [comment]
That should fix it. Also made it so that the surplus armor vendor gives /obj/item/clothing/head/helmet/standard instead of /obj/item/clothing/head/helmet since they're the same thing anyway.

**Review by SplinterGP — CHANGES_REQUESTED** [review]
Too much duplicated code

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Dupe code, missing Changelog and code churn in code/game/objects/machinery/vending/marine_vending.dm

**Pull request closed without merging**